src/Features/LanguageServer/Protocol/Handler/InlineCompletions/InlineCompletionsHandler.cs

```diff
@@ -0,0 +1,288 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Collections;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Snippets;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler.InlineCompletions;
+
+/// <summary>
+/// Supports built in legacy snippets for razor scenarios.
+/// </summary>
+[ExportRoslynLanguagesLspRequestHandlerProvider, Shared]
+[ProvidesMethod(VSInternalMethods.TextDocumentInlineCompletionName)]
+internal partial class InlineCompletionsHandler : AbstractStatelessRequestHandler<VSInternalInlineCompletionRequest, VSInternalInlineCompletionList>
+{
+    /// <summary>
+    /// The set of built in snippets from, typically found in
+    /// C:\Program Files\Microsoft Visual Studio\2022\VS_INSTANCE\VC#\Snippets\1033\Visual C#
+    /// These are currently the only snippets supported.
+    /// </summary>
+    public static ImmutableHashSet<string> BuiltInSnippets = ImmutableHashSet.Create(
```

Comment:
I've limited these to built in snippets for now.  While the parsing could probably handle most custom snippets, there are a few things that would not work, namely snippets with additional references (requires lsp response to support additional text edits) and selection (currently snippets are only invocable via tab-tab, so selection does not make sense in that context).

Depending on the direction that the editor takes xml snippets and where we go with semantic snippets, we can potentially open this up to more with a few improvements on the lsp api.